src/globus_cli/commands/gcs/endpoint/role/list.py

```diff
@@ -0,0 +1,42 @@
+import uuid
+
+import click
+from globus_sdk.paging import Paginator
+
+from globus_cli.commands.gcs.endpoint.role._common import role_fields
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import command, endpoint_id_arg
+from globus_cli.services.transfer import iterable_response_to_dict
```

Comment:
It feels slightly awkward / boundary-breaking to me that we'd have a non-Transfer-API command which pulls in a data handler from Transfer tools. On the other hand, I can understand why you'd prefer not to make things different unnecessarily, and the helper doesn't _really_ have anything to do with Transfer (other than aping its payload format a little).

Since we're already putting some supporting tools into `utils`, maybe we should do one more in this PR? Here's a slightly revamped version of this helper which might be nice:
```python
def iterable_response_to_dict(iterator: t.Iterable[t.Any], *, key: str = "DATA") -> dict[str, list[t.Any]]:
    output_dict: dict[str, list[t.Any]] = {key: []}
    for item in iterator:
        try:
            value = item.data
        except AttributeError:
            value = item
        output_dict[key].append(value)
    return output_dict
```

What do you think?

Reply:
Sure, I'm happy with the rewrite.

One disagreement, I don't like using a generic `utils.py`. I did it in this PR because I couldn't think of a more specific place to put a common function like `resolve_principal_urn`.

This function however does have a specific usage pattern, which is termio formatting (there are 7 usage sites across the CLI, all of which are calls to `display(..., json_converter=iterable_response_to_dict)`), so I'd prefer to put it in the `termio` package.

Things that grow together go together.

Reply:
Okay, I didn't rewrite but I removed the transfer import.

While moving this stuff around, I noticed that somebody (2022-sirosen) had added [basically duplicate functionality](https://github.com/globus/globus-cli/blob/main/src/globus_cli/utils.py#L153-L164) in the PagingWrapper itself so I used that instead.

The only notable distinction is that transfer's `iterable_response_to_dict` attempts to extract data from a `data` attribute from the item falling back to the item itself if a `data` attribute doesn't exist. For this usage, a `data` attribute doesn't exist so it's identical.

Reply:
> One disagreement, I don't like using a generic `utils.py` ...

I disagree with your disagreement! We actually agree! :upside_down_face: 

I view `utils.py` (and worse, `utils/`) as a code-smell in many cases. It too often represents "I've given up on namespacing this thing."
Waaay back in time I used to be a bit more of a maniac about it and try to actually eliminate a `utils.py` module from codebases. I've since grown more lenient -- especially for things like "functions I wish were in the stdlib or builtins in the context of this codebase." But there's a danger of growing too lax and starting to just dump your whole application into there. So, yeah, if we want output munging tools, `termio` is a nice place for them.


> I noticed that somebody (2022-sirosen)

I never did like that guy much. :stuck_out_tongue_winking_eye: 

I'm good with the way you rewrote it. We can, as we agree, iterate further in the future.